### PR TITLE
Call pre-build-dockerfile before docker build

### PR DESCRIPTION
**Pull request by 3onyc**

### CODE REVIEW

File: plugins/00_dokku-standard/commands [hunk 1]
```diff
@@ -38,10 +38,12 @@ case "$1" in
         DOCKERFILE_PORT=$(grep EXPOSE Dockerfile | head -1 | awk '{ print $2 }' || true)
         [[ -n "$DOCKERFILE_PORT" ]] && dokku config:set-norestart $APP DOKKU_DOCKERFILE_PORT=$DOCKERFILE_PORT
 
-        # sticking with same pattern of building app image before pre-build for now.
-        docker build -t "$IMAGE" .
         # buildstep pluginhooks don't necessarily make sense for dockerfiles. call the new breed!!!
         pluginhook pre-build-dockerfile "$APP"
+
+        # sticking with same pattern of building app image before pre-build for now.
```
Comment (michaelshobbs) [inline plugins/00_dokku-standard/commands]: This comment should probably be removed.

**Comment by michaelshobbs** [comment]
This was (loosely) intentionally called in the same manner as buildstep deploys to maintain a similar execution path. Can you provide your use case that requires this change?

**Comment by 3onyc** [comment]
I don't have an example use-case right now as I noticed this while reading the code.

However, looking at the logic for buildstep, the [actual build](https://github.com/progrium/dokku/blob/c2053d532277eb00b1db074bebd203af12e0136e/plugins/00_dokku-standard/commands#L26) is triggered after the `pre-build*` hooks.

**Comment by michaelshobbs** [comment]
Hm. I'm clearly not recalling correctly. Seems legit to me then.

File: plugins/00_dokku-standard/commands (hunk 1 above)
Comment (3onyc) [inline plugins/00_dokku-standard/commands]: Done!

**Comment by michaelshobbs** [comment]
@josegonzalez: what do you think here? 

Dockerfile deployments are brand new for dokku and I don't think there are any dockerfile-specific plugins yet. I was hoping to feedback on this workflow as folks started to use it. In any case, as this is early days making this particular change is probably semantically correct at the least. :smile:

**Comment by josegonzalez** [comment]
Seems legit to me.

**Pull request merged**